videostore/utils.go

```diff
@@ -305,20 +370,20 @@ func validateTimeRange(files []string, start, end time.Time) error {
 	return nil
 }
 
-// getVideoDuration returns the duration of the video file in seconds.
-func getVideoDuration(filePath string) (time.Duration, error) {
+// getVideoInfo calls the C function get_video_info to retrieve
+// duration, width, height, and codec of a video file.
+func getVideoInfo(filePath string) (videoInfo, error) {
 	cFilePath := C.CString(filePath)
 	defer C.free(unsafe.Pointer(cFilePath))
-
-	var duration C.int64_t
-	ret := C.get_video_duration(&duration, cFilePath)
+	var cinfo C.VideoInfo
+	ret := C.get_video_info(&cinfo, cFilePath)
 	switch ret {
-	case C.VIDEO_STORE_DURATION_RESP_OK:
-		// Convert duration from AV_TIME_BASE units to time.Duration
-		return time.Duration(duration) * time.Microsecond, nil
-	case C.VIDEO_STORE_DURATION_RESP_ERROR:
-		return 0, fmt.Errorf("failed to get video duration for file: %s", filePath)
+	case C.VIDEO_STORE_VIDEO_INFO_RESP_OK:
+		return fromCVideoInfo(cinfo), nil
+	case C.VIDEO_STORE_VIDEO_INFO_RESP_ERROR:
+		return videoInfo{}, fmt.Errorf("get_video_info failed for file: %s", filePath)
```

Comment:
why is the specific case error more general than the general error? Is it because it's one of our own custom errors?

Reply:
The default case is used to bubble up libav error codes, whereas VIDEO_STORE_VIDEO_INFO_RESP_ERROR is a catchall for other error cases that do not derive from libav errors.